pkg/utils/grpc.go

```diff
@@ -32,19 +32,23 @@ const DefaultRPCTimeout = 60 * time.Second
 var RPCTimeout = DefaultRPCTimeout
 
 // CreateGrpcConnection create a grpc connection with given port
-func CreateGrpcConnection(ctx context.Context, c client.Client, pod *v1.Pod, port int) (*grpc.ClientConn, error) {
-	nodeName := pod.Spec.NodeName
-	log.Info("Creating client to chaos-daemon", "node", nodeName)
+func CreateGrpcConnection(ctx context.Context, c client.Client, pod *v1.Pod, port int, address string) (*grpc.ClientConn, error) {
```

Comment:
`pod` and `address` will not used at same time. `pod` has some extra logic to find out actual (node) address.

How about refactor this methods? Writing another “overloading” method, like
`CreateGrpcConnection(ctx,c,pod,port)` and `CreateGrpcConnectionWithAddress(ctx,c,address,port)`

Reply:
thx for the suggestion! fixed